### DISCUSSION TITLE
MiniSurround: autodetect nearest surrounding like sandwich `iss`

### DISCUSSION THREAD

**Discussion by sunaku** [nvim-mini/mini.nvim#450] [Q&A] [answered]
Hello,

Sandwich provides special [`iss` and `ass` text objects](https://github.com/machakann/vim-sandwich/wiki/Introduce-vim-surround-keymappings#text-objects-to-select-the-nearest-surrounded-text-automatically) that will automatically detect the nearest surround delimiters.  How can this feature be replicated in MiniSurround?

I tried using `custom_surroundings` but, per my limited understanding, it appears to target a single expression per surrounding.  For example, to replicate Sandwich's behavior, I would need a custom surrounding that loops through several different possible delimiters (all kinds of quotation marks first, followed by all kinds of brackets) and stops at the first successful match.

```lua
require("mini.surround").setup({
  custom_surroundings = {
    ["s"] = {
      input = { find = "[\"'`{(\\[<].-[\"'`})\\]>]", extract = "^(.).*(.)$" },
    },
  },
})
```

Thanks for your consideration.

**Comment by echasnovski** [comment, marked as answer]
Yes, this is not entirely possible at the moment. Constructing custom surrounding seems like the best course of action here.

By the way, you seem to use not the most recent version of 'mini.surround', because this format for `custom_surroundings` was replaced with the new one in 'mini.nvim' 0.6.0.
